models/groups.go

```diff
@@ -25,38 +25,38 @@ var (
 	groupPrecedenceMax    = int64(100)
 )
 
-//Type to map for the Cognito GroupType object
+//Group is a type to map for the Cognito GroupType object
```

Comment:
This wording confused me a little (and I realise this is existing wording here). This may just be me, but perhaps rewording to remove the mention of Cognito here could be more clear:

```suggestion
//Group is a type for the identity API representation of a group's details
```

I realise that this is also true of comments throughout the models that are not always clear that they are identity API models as opposed to Cognito API models due to references to Cognito. If we can avoid referencing Cognito here then that would be better since it may be that in the future we decide to swap out Cognito as the storage backend and implement our own using mongo for example so being clear what are API models means that we are clear on what the unchanging external interface to this microservice is.